src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -718,54 +698,52 @@ where
             ..Default::default()
         };
 
-        let internal_tables = match self
+        let res = self
             .create_relation_on_compute_node(
                 &Relation::Table(mview.clone()),
                 fragment_graph,
                 mview_id,
                 &mut ctx,
             )
-            .await
-        {
-            Err(e) => {
-                let internal_table_ids = ctx
-                    .internal_table_id_map
-                    .keys()
-                    .cloned()
-                    .chain(std::iter::once(mview_id))
-                    .collect_vec();
+            .await;
 
-                self.stream_manager
-                    .remove_processing_table(internal_table_ids, true)
-                    .await;
+        if let Err(err) = res {
+            self.stream_manager
+                .remove_processing_table(
+                    ctx.internal_table_ids()
+                        .into_iter()
+                        .chain(std::iter::once(mview_id))
+                        .collect_vec(),
+                    true,
+                )
+                .await;
```

Comment:
I've no idea why there's duplication between this function and `create_relation`. Can we merge them together? cc @nanderstabel

Reply:
Will try to do some refactor on ddl service.

Reply:
@yezizp2012 did you already start with this? Else I will go and fix it.